xarray/core/treenode.py

```diff
@@ -560,7 +560,10 @@ def __delitem__(self: Tree, key: str):
             del self._children[key]
             child.orphan()
         else:
-            raise KeyError("Cannot delete")
+            try:
+                raise ValueError("cannot delete key not found in child nodes")
+            except ValueError as e:
+                raise KeyError(key) from e
```

Comment:
implementing suggestion made in https://github.com/pydata/xarray/pull/9222/files#r1674282192

```suggestion
            raise KeyError(key) from ValueError("cannot delete key not found in child nodes")
```